internal/funcutil/collections.go

```diff
@@ -160,6 +160,16 @@ func Contains[T comparable](a []T, x T) bool {
 	return Exists(a, func(y T) bool { return x == y })
 }
 
+// Set takes an array of elements and returns a set of those elements, represented as a map from
+// elements to empty struct
+func Set[T comparable](a []T) map[T]any {
```

Comment:
Why is the resulting type `map[T]any` instead of `map[T]struct{}`?

Reply:
It should be `map[T]struct{}`